File: pueblo_golf_tournament_mobile/lib/dto/request/register-payment-request-dto.dart
import 'dart:io';
import 'dart:typed_data';

import 'package:file_picker/file_picker.dart';
import 'package:json_annotation/json_annotation.dart';

class RegisterPaymentRequestDto {
  final String paymentMethod;
  final int registrationId;
  final String referrenceId;
  final int paymentChannelAccountId;
  final String accountNumber;
  final String accountName;
  final DateTime paymentDate;
  final File paymentReferrencePhoto;

  RegisterPaymentRequestDto({
    required this.paymentChannelAccountId,
    required this.accountNumber,
    required this.accountName,
    required this.paymentMethod,
    required this.registrationId,
    required this.referrenceId,
    required this.paymentDate,
    required this.paymentReferrencePhoto,
  });
}

class RegisterPaymentWebRequestDto {
  final String paymentMethod;
  final int registrationId;
  final String referrenceId;
  final DateTime paymentDate;
  final int paymentChannelAccountId;
  final String accountNumber;
  final String accountName;
  final Uint8List paymentReferrencePhoto;

  RegisterPaymentWebRequestDto({
    required this.paymentChannelAccountId,
    required this.accountNumber,
    required this.accountName,
    required this.paymentMethod,
    required this.registrationId,
    required this.referrenceId,
    required this.paymentDate,
    required this.paymentReferrencePhoto,
  });
}

File: pueblo-golf-tournament-api/Dto/Incoming/RegisterPaymentRequestDto.cs
namespace pueblo_golf_tournament_api.Dto.Incoming
{
    public class RegisterPaymentRequestDto
    {
        public long RegistrationId { get; set; }
        public string? PaymentMethod { get; set; }
        public long PaymentChannelAccountId { get; set; }
        public string? AccountNumber { get; set; }
        public string? AccountName { get; set; }
        public string? ReferrenceId { get; set; }
        public IFormFile PaymentReferrencePhoto { get; set; }
    }
}